scrapy/responsetypes.py

```diff
@@ -86,34 +92,33 @@ def from_filename(self, filename):
         else:
             return Response
 
-    def from_body(self, body):
-        """Try to guess the appropriate response based on the body content.
-        This method is a bit magic and could be improved in the future, but
-        it's not meant to be used except for special cases where response types
-        cannot be guess using more straightforward methods."""
-        chunk = body[:5000]
-        chunk = to_bytes(chunk)
-        if not binary_is_text(chunk):
-            return self.from_mimetype('application/octet-stream')
-        elif b"<html>" in chunk.lower():
-            return self.from_mimetype('text/html')
-        elif b"<?xml" in chunk.lower():
-            return self.from_mimetype('text/xml')
-        else:
-            return self.from_mimetype('text')
```

Comment:
Please, refrain from removing any existing (public) method. We need to keep them around and working as before for backward compatibility, to avoid existing user code that may be using them.

Instead, log a deprecation warning at the beginning of each one, asking users to use `from_args` instead. Look for `DeprecationWarning` usage in the code base to get an idea on how to use it, and make sure you use the parameter of the warn function that makes the warning be about the calling code.